lib/api/src/sys/function_env.rs

```diff
@@ -117,6 +117,17 @@ impl<T: Send + 'static> FunctionEnvMut<'_, T> {
             func_env: self.func_env.clone(),
         }
     }
+
+    /// Borrows a new mutable reference of both the attached Store and host state
+    pub fn data_and_store_mut(&mut self) -> (&mut T, StoreMut) {
+        let data = self.func_env.as_mut(&mut self.store_mut) as *mut T;
+        // telling the borrow check to close his eyes here
+        // this is still relatively safe to do as func_env are
+        // stored in a specific vec of Store, separate from the other objects
+        // and not really directly accessible with the StoreMut
+        let data = unsafe { &mut *data };
+        (data, self.store_mut.as_store_mut())
+    }
```

Comment:
Can we add it on the js version as well and add a `#[universal_test]` verifying the desired behavior? (that way, we test that it also works in js)

Reply:
It is on js (in the 2nd commit).
I'll find something for a test.

Reply:
Oh I missed the js impl. Let's just have an universal test